master/internal/workspace/authz_iface.go

```diff
@@ -16,6 +16,10 @@ type WorkspaceAuthZ interface {
 		ctx context.Context, curUser model.User, workspace *workspacev1.Workspace,
 	) error
 
+	CanGetWorkspaceID(
```

Comment:
if we can just use `CanGetWorkspace`, I'd prefer that and not create a new authz method